Automattic/wp-calypso
# Reader stream - revert to 'magic' item selection if selected item is out of the render group.

Related to # Automattic/loop#181

## Proposed Changes

* Updates the reader stream's  `selectNextItem` method to fallback to the magic select functionality if the currently selected item is not rendered (cannot be scrolled to).
* This is a bit of an edge case, but fixes a bug that makes shortkeys feel broken in some situations.

## Why are these changes being made?


* The "j" shorkey fails to scroll to the next selected item if the currently selected item is far enough away in scroll position to no longer be rendered by the InfiniteList component. This makes it seem like shortkeys are broken and not functioning.
* There is pre-existing functionality for selecting a next item based on what is visible. We fall back to this logic instead in the rare case that the previously selected item is no longer rendered.

## Testing Instructions



* Run this calypso branch.
* Select a post with "j"
* press "j" once more and verify it selects the next item.
* scroll down a tiny bit, ensure "j" selects the next item in the list.
* scroll down a lot. ensure "j" selects an item based on what is visible - previously here is where pressing "j" would feel like it did absolutely nothing.

## Pre-merge Checklist
